diff --git a/drivers/net/ixgbe/ixgbe_ethdev.c b/drivers/net/ixgbe/ixgbe_ethdev.c
index a2bc708..c841526 100644
--- a/drivers/net/ixgbe/ixgbe_ethdev.c
+++ b/drivers/net/ixgbe/ixgbe_ethdev.c
@@ -4442,12 +4442,22 @@ ixgbe_dev_promiscuous_enable(struct rte_eth_dev *dev)
 {
 	struct ixgbe_hw *hw = IXGBE_DEV_PRIVATE_TO_HW(dev->data->dev_private);
 	uint32_t fctrl;
+	uint32_t rxctrl;
+
+	/* Before modifying FCTRL, RXCTRL.RXEN must be 0 */
+	rxctrl = IXGBE_READ_REG(hw, IXGBE_RXCTRL);
+	rxctrl &= ~IXGBE_RXCTRL_RXEN;
+	IXGBE_WRITE_REG(hw, IXGBE_RXCTRL, rxctrl);
 
 	fctrl = IXGBE_READ_REG(hw, IXGBE_FCTRL);
 	fctrl |= (IXGBE_FCTRL_UPE | IXGBE_FCTRL_MPE);
 	IXGBE_WRITE_REG(hw, IXGBE_FCTRL, fctrl);
+
+	/* Re-enable RXEN now */
+	rxctrl |= IXGBE_RXCTRL_RXEN;
+	IXGBE_WRITE_REG(hw, IXGBE_RXCTRL, rxctrl);
 
	return 0;
 }
 
 static void
@@ -4455,6 +4465,12 @@ ixgbe_dev_promiscuous_disable(struct rte_eth_dev *dev)
 {
 	struct ixgbe_hw *hw = IXGBE_DEV_PRIVATE_TO_HW(dev->data->dev_private);
 	uint32_t fctrl;
+	uint32_t rxctrl;
+
+	/* Before modifying FCTRL, RXCTRL.RXEN must be 0 */
+	rxctrl = IXGBE_READ_REG(hw, IXGBE_RXCTRL);
+	rxctrl &= ~IXGBE_RXCTRL_RXEN;
+	IXGBE_WRITE_REG(hw, IXGBE_RXCTRL, rxctrl);
 
 	fctrl = IXGBE_READ_REG(hw, IXGBE_FCTRL);
 	fctrl &= (~IXGBE_FCTRL_UPE);
@@ -4463,8 +4479,12 @@ ixgbe_dev_promiscuous_disable(struct rte_eth_dev *dev)
 	else
 		fctrl &= (~IXGBE_FCTRL_MPE);
 	IXGBE_WRITE_REG(hw, IXGBE_FCTRL, fctrl);
 
+	/* Re-enable RXEN now */
+	rxctrl |= IXGBE_RXCTRL_RXEN;
+	IXGBE_WRITE_REG(hw, IXGBE_RXCTRL, rxctrl);
+
	return 0;
 }
 
 static void
@@ -4472,11 +4492,21 @@ ixgbe_dev_allmulticast_enable(struct rte_eth_dev *dev)
 {
 	struct ixgbe_hw *hw = IXGBE_DEV_PRIVATE_TO_HW(dev->data->dev_private);
 	uint32_t fctrl;
+	uint32_t rxctrl;
+
+	/* Before modifying FCTRL, RXCTRL.RXEN must be 0 */
+	rxctrl = IXGBE_READ_REG(hw, IXGBE_RXCTRL);
+	rxctrl &= ~IXGBE_RXCTRL_RXEN;
+	IXGBE_WRITE_REG(hw, IXGBE_RXCTRL, rxctrl);
 
 	fctrl = IXGBE_READ_REG(hw, IXGBE_FCTRL);
 	fctrl |= IXGBE_FCTRL_MPE;
+
+	/* Re-enable RXEN now */
+	rxctrl |= IXGBE_RXCTRL_RXEN;
+	IXGBE_WRITE_REG(hw, IXGBE_RXCTRL, rxctrl);
 	IXGBE_WRITE_REG(hw, IXGBE_FCTRL, fctrl);
 
	return 0;
 }
 
@@ -4485,15 +4515,25 @@ ixgbe_dev_allmulticast_disable(struct rte_eth_dev *dev)
 {
 	struct ixgbe_hw *hw = IXGBE_DEV_PRIVATE_TO_HW(dev->data->dev_private);
 	uint32_t fctrl;
+	uint32_t rxctrl;
 
 	if (dev->data->promiscuous == 1)
 		return; /* must remain in all_multicast mode */
 
+	/* Before modifying FCTRL, RXCTRL.RXEN must be 0 */
+	rxctrl = IXGBE_READ_REG(hw, IXGBE_RXCTRL);
+	rxctrl &= ~IXGBE_RXCTRL_RXEN;
+	IXGBE_WRITE_REG(hw, IXGBE_RXCTRL, rxctrl);
+
 	fctrl = IXGBE_READ_REG(hw, IXGBE_FCTRL);
 	fctrl &= (~IXGBE_FCTRL_MPE);
 	IXGBE_WRITE_REG(hw, IXGBE_FCTRL, fctrl);
+
+	/* Re-enable RXEN now */
+	rxctrl |= IXGBE_RXCTRL_RXEN;
+	IXGBE_WRITE_REG(hw, IXGBE_RXCTRL, rxctrl);
 
	return 0;
 }
 
 /**
